semgrep-core/src/tainting/Dataflow_tainting.mli

```diff
@@ -10,6 +10,7 @@ type overlap = float
 type propagator_id = var
 type propagator_from = propagator_id
 type propagator_to = propagator_id
+type taint_info = Tainted of Taint.taints | MarkedClean
```

Comment:
From a memory perspective, it may be more efficient to have:
```ocaml
type var_env = {
  taints : Taint.taints;
  clean : VarSet.t;
}
```
Otherwise we are adding an extra word overhead for every variable in the map. In general it doesn't matter, except when we encounter some large function or machine generated code. Although we shouldn't get that code to be analyzed in the first place, but we still do...

Reply:
I switched to remove variables in the non-dot case, so I think your suggestion would now be equivalent to what we have

Reply:
It's not really the same thing, I would say they are unrelated issues. So previously we had a mapping from `var` to `Taint.taints`. Now the mapping is from `var` to `taint_info`. So we are wrapping `Taint.taints` with an extra constructor `Tainted`. That extra constructor means memory overhead, and it's an overhead we will pay for every `var` in the environment. So N `var`s means N times that overhead, and in the worst case scenario you would have to multiply N by the number of mappings that we have, and we have two for each node in the CFG.

WIth
```ocaml
type var_env = {
  taints : Taint.taints;
  clean : VarSet.t;
}
```
the overhead is fixed wrt the number of variables in the mapping. Although it's still related to the number of nodes in the CFG.

That said no need to do anything about this now, I can think on this myself later on. I just wanted to explain better my comment. But correct me if you think my reasoning is wrong here.

Hope you don't take it bad that I'm nitpicking on these memory details!

Reply:
I wasn't thinking right when I proposed the solution. Although the problem is real. I need to think if there is another way.